modules/mbedtls/crypto_mbedtls.cpp

```diff
@@ -384,3 +460,21 @@ Vector<uint8_t> CryptoMbedTLS::decrypt(Ref<CryptoKey> p_key, Vector<uint8_t> p_c
 	copymem(out.ptrw(), buf, size);
 	return out;
 }
+
+PackedByteArray CryptoMbedTLS::hmac_digest(HashingContext::HashType p_hash_type, PackedByteArray p_key, PackedByteArray p_msg) {
+	int len = 0;
+	mbedtls_md_type_t ht = CryptoMbedTLS::md_type_from_hashtype(p_hash_type, len);
+
+	bool allowed = HMACContextMbedTLS::is_md_type_allowed(ht);
+
+	ERR_FAIL_COND_V_MSG(!allowed, PackedByteArray(), "Unsupported hash type.");
+
+	const mbedtls_md_info_t *info = mbedtls_md_info_from_type((mbedtls_md_type_t)ht);
+
+	PackedByteArray out;
+	out.resize(len);
+
+	unsigned char *out_ptr = (unsigned char *)out.ptr();
+	int ret = mbedtls_md_hmac(info, (const unsigned char *)p_key.ptr(), (size_t)p_key.size(), (const unsigned char *)p_msg.ptr(), (size_t)p_msg.size(), out_ptr);
```

Comment:
I would probably use the HMACContext class instead, and move this function in `core/crypto.h`. Just make sure to check that `HMACContext::create() != nullptr`.

Reply:
I originally planned to do this, but I decided to follow the lead of the existing methods, which seemed to not use the Context classes. Happy to change.